pkg/plugins/manager/client/decorator.go

```diff
@@ -123,4 +130,9 @@ func reverseMiddlewares(middlewares []plugins.ClientMiddleware) []plugins.Client
 	return reversed
 }
 
+func instrumentContext(ctx context.Context, endpoint string, pCtx backend.PluginContext) context.Context {
```

Comment:
With the approach you did choose I think you can keep this in instrumentation.go for now as you had initially.

Reply:
With that I needed to change the function signature in e.g. `InstrumentCallResourceRequest` to also contain the context:https://github.com/grafana/grafana/pull/74428/files#diff-f68ce913764de9b2ec9c118f29476e2aeec3981d7561124e48659e5a8c2c6f78L111

Let me know what you think.